File: python/Excessive_data_exposure.py
def get_user_data(user_id):
    # retrieve user data from database
    user = User.objects.get(id=user_id)
    return user.to_dict()

@app.route('/users/<int:user_id>', methods=['GET'])
def expose_user_data(user_id):
    user_data = get_user_data(user_id)
    return user_data
  

File: java/Excessive_data_exposure.java
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@SpringBootApplication
@RestController
public class UserDataController {

    @GetMapping("/users/{userId}")
    public User getUserData(@PathVariable("userId") int userId) {
        // retrieve user data from database
        User user = userRepository.findById(userId);
        return user;
    }

    public static void main(String[] args) {
        SpringApplication.run(UserDataController.class, args);
    }
}
